base/applications/notepad/dialog.c

```diff
@@ -1117,82 +1117,79 @@ VOID DIALOG_Replace(VOID)
     DIALOG_SearchDialog(ReplaceText);
 }
 
+typedef struct tagGOTO_DATA
+{
+    INT iLine;
+    INT cLines;
+} GOTO_DATA, *PGOTO_DATA;
+
 static INT_PTR
 CALLBACK
 DIALOG_GoTo_DialogProc(HWND hwndDialog, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
-    BOOL bResult = FALSE;
-    HWND hTextBox;
-    TCHAR szText[32];
+    static PGOTO_DATA s_pGotoData;
 
-    switch(uMsg) {
-    case WM_INITDIALOG:
-        hTextBox = GetDlgItem(hwndDialog, ID_LINENUMBER);
-        _sntprintf(szText, ARRAY_SIZE(szText), _T("%Id"), lParam);
-        SetWindowText(hTextBox, szText);
-        break;
-    case WM_COMMAND:
-        if (HIWORD(wParam) == BN_CLICKED)
+    switch (uMsg)
+    {
+        case WM_INITDIALOG:
+            s_pGotoData = (PGOTO_DATA)lParam;
+            SetDlgItemInt(hwndDialog, ID_LINENUMBER, s_pGotoData->iLine, FALSE);
+            return TRUE; /* Set focus */
+
+        case WM_COMMAND:
         {
             if (LOWORD(wParam) == IDOK)
             {
-                hTextBox = GetDlgItem(hwndDialog, ID_LINENUMBER);
-                GetWindowText(hTextBox, szText, ARRAY_SIZE(szText));
-                EndDialog(hwndDialog, _ttoi(szText));
-                bResult = TRUE;
+                INT iLine = GetDlgItemInt(hwndDialog, ID_LINENUMBER, NULL, FALSE);
+                // TODO: Should we display error message?
```

Comment:
What error message? If we want to go further than the end of file?
That would be the only reason for the "total number of lines" (cLines) to be passed via the goto_data structure.

Reply:
No error message is better?

Reply:
Personally an error would be better. Windows' notepad has one too: "The line number is larger than the total number of lines." (something like that; it looks as follows in french, I don't have an english version to see the actual error message)
![image](https://user-images.githubusercontent.com/1969829/219907081-e79fdbe5-86aa-422c-b2f6-c69dac63c1b5.png)

Reply:
Fixed.

Reply:
![image](https://user-images.githubusercontent.com/610685/220066880-2695913f-981e-43c8-b6c8-d521950a20b4.png)

Reply:
The line number zero is out of range.

Reply:
Zero and a too big number display the same text on the notepad on Win10.